From dd1fb49e8a0ae6124c1388a5a486f9d9cb3a053b Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Mon, 08 Jul 2024 18:07:58 +0300
Subject: [PATCH] [FSW-420] fs1: check valid firmware version

with this, added support to verify firmware
version lies between max firmware version and
minimum version defined by security data.

Change-Id: Id2ce9d49ccbfa4f8b8aaba00972da5d55fe6521e
---

diff --git a/common/security.c b/common/security.c
index 08b4de9..787a75a 100644
--- a/common/security.c
+++ b/common/security.c
@@ -8,6 +8,17 @@
 #include <boot_rom.h>
 #include <dev_security.h>
 #include <string.h>
+#include <security.h>
+
+int security_check_valid_fw_version(uint32_t fw_version)
+{
+	if ((fw_version < boot_rom_global.sec_data.sec_ver) ||
+		(fw_version > SECURITY_MAX_FW_SEC_VERSION)) {
+		return SECURITY_FW_VERSION_NOT_SECURE;
+	}
+
+	return SECURITY_FW_VERSION_SECURE;
+}
 
 int security_check_key_revoke_status(uint32_t key_index)
 {
diff --git a/common/security.h b/common/security.h
index 58b9354..1d03c35 100644
--- a/common/security.h
+++ b/common/security.h
@@ -6,8 +6,13 @@
 #ifndef _SECURITY_H_
 #define _SECURITY_H_
 
+#define SECURITY_FW_VERSION_SECURE		1
+#define SECURITY_FW_VERSION_NOT_SECURE		-1
+#define SECURITY_MAX_FW_SEC_VERSION		128
+
 void boot_rom_security_init(void);
 int security_check_key_revoke_status(uint32_t key_index);
 int security_rom_get_key_hash_by_idx(uint32_t *hash, uint8_t key_index);
+int security_check_valid_fw_version(uint32_t fw_version);
 
 #endif /* _SECURITY_H_ */
